src/highdicom/seg/sop.py

```diff
@@ -940,3 +883,18 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
                 return pack_bits(planes.flatten())
             else:
                 return planes.flatten().tobytes()
+
+    def add_segments(
+        self,
+        pixel_array: np.ndarray,
+        segment_descriptions: Sequence[SegmentDescription],
+        plane_positions: Optional[Sequence[PlanePositionSequence]] = None,
+        omit_empty_frames: bool = True,
+    ) -> None:
+        raise DeprecationWarning(
+            'To ensure correctness of segmentation images, the add_segments '
+            'method was deprecated in highdicom 0.8.0. For more information '
+            'and migration instructions visit '
+            'https://highdicom.readthedocs.io/en/latest/release_notes.html'
+            '#deprecation-of-add-segments-method'
+        )
```

Comment:
Hmm good point. I do think having some sort of useful error message here with a clue as to what has happened will be very useful to people though. In 1c03498 I switched to `NotImplementedError`, what do you think about that?

Reply:
`NotImplementedError` is typically used to indicate that a method is not implemented on class derived from some abstract class base. I would just raise an `AttributeError`. The same Exception type would be raise if the method wouldn't exist.

Reply:
I think this is nicer, because we can then entirely remove the method in the future without breaking anyone's code (in case someone may handle the raised `NotImplementedError`).